src/torchmetrics/classification/cohen_kappa.py

```diff
@@ -15,11 +15,173 @@
+
+    where :math:`p_o` is the empirical probability of agreement and :math:`p_e` is
+    the expected agreement when both annotators assign labels randomly. Note that
+    :math:`p_e` is estimated using a per-annotator empirical prior over the
+    class labels.
+
+    Accepts the following input tensors:
+
+    - ``preds``: ``(N, ...)`` (int tensor) or ``(N, C, ..)`` (float tensor). If preds is a floating point
+      we apply ``torch.argmax`` along the ``C`` dimension to automatically convert probabilities/logits into
+      an int tensor.
+    - ``target`` (int tensor): ``(N, ...)``
+
+    Additional dimension ``...`` will be flattened into the batch dimension.
+
+    Args:
+        num_classes: Integer specifing the number of classes
+        ignore_index:
+            Specifies a target value that is ignored and does not contribute to the metric calculation
+        weights: Weighting type to calculate the score. Choose from:
+
+            - ``None`` or ``'none'``: no weighting
+            - ``'linear'``: linear weighting
+            - ``'quadratic'``: quadratic weighting
+        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
+            Set to ``False`` for faster computations.
+        kwargs: Additional keyword arguments, see :ref:`Metric kwargs` for more info.
+
+    Example (pred is integer tensor):
+        >>> from torchmetrics import MulticlassCohenKappa
+        >>> target = torch.tensor([2, 1, 0, 0])
+        >>> preds = torch.tensor([2, 1, 0, 1])
+        >>> metric = MulticlassCohenKappa(num_classes=3)
+        >>> metric(preds, target)
+        tensor(0.6364)
+
+    Example (pred is float tensor):
+        >>> from torchmetrics import MulticlassCohenKappa
+        >>> target = torch.tensor([2, 1, 0, 0])
+        >>> preds = torch.tensor([
+        ...   [0.16, 0.26, 0.58],
+        ...   [0.22, 0.61, 0.17],
+        ...   [0.71, 0.09, 0.20],
+        ...   [0.05, 0.82, 0.13],
+        ... ])
+        >>> metric = MulticlassCohenKappa(num_classes=3)
+        >>> metric(preds, target)
+        tensor(0.6364)
+
+    """
+    is_differentiable: bool = False
+    higher_is_better: bool = True
+    full_state_update: bool = False
+
+    def __init__(
+        self,
+        num_classes: int,
+        ignore_index: Optional[int] = None,
+        weights: Optional[Literal["linear", "quadratic", "none"]] = None,
+        validate_args: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(num_classes, ignore_index, normalize=None, validate_args=False, **kwargs)
+        if validate_args:
+            _multiclass_cohen_kappa_arg_validation(num_classes, ignore_index, weights)
+        self.weights = weights
+        self.validate_args = validate_args
+
+    def compute(self) -> Tensor:
+        return _cohen_kappa_reduce(self.confmat, self.weights)
+
+
+# -------------------------- Old stuff --------------------------
+
+
 class CohenKappa(Metric):
```

Comment:
shall we add some future warning to notify users/developers that they should rather use Binary/M-class versions?